apps/03TicTacToe.js

```diff
@@ -24,23 +24,81 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+
+//  Your code here
+  if ((board[0][0] === 'X' && board[0][1] === 'X' && board[0][2] === 'X' ) || (board[0][0] === 'O' && board[0][1] === 'O' && board[0][2] === 'O')) {
+      console.log(playerTurn + " !!! WINS!!!    ...Resetting Game");
+
+        board = [
+          [' ', ' ', ' '],
+          [' ', ' ', ' '],
+          [' ', ' ', ' ']
+        ];
+return true;
+  } else
+  return false;
 }
 
 function verticalWin() {
-  // Your code here
+        if  ((board[0][0] === 'X' && board[1][0] === 'X' && board[2][0] === 'X' )
```

Comment:
The rationale was that a user can put in "a", "0", "$", etc..    Since I know it's X or Y turn I took that ability away from them.  As long as it was their turn I infer that it's X or O.  I guess i failed to see Spec  3.